.github/workflows/CI.yml

```diff
@@ -218,20 +213,68 @@ jobs:
+          cp npm-generated-sources.json build/npm-generated-sources.json
+          mv npm-generated-sources.json npm/npm-generated-sources.json
+          mv package-lock.json npm/package-lock.json
+
+      - name: Checkout Flathub Shared Modules
+        uses: actions/checkout@v3
+        with:
+          repository: flathub/shared-modules
+          path: build/shared-modules
+
+      - name: External Data Checker
+        run: |
+          flatpak run org.flathub.flatpak-external-data-checker build/dev.lizardbyte.sunshine.yml
+
       - name: Build Linux Flatpak
         working-directory: build
         run: |
-          sudo su $(whoami) -c 'flatpak run org.flatpak.Builder --arch=${{ matrix.arch }} --repo=repo --force-clean \
-            --stop-at=cuda build-sunshine dev.lizardbyte.sunshine.yml'
+          mkdir -p ~/.local/bin
+          wget https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64 -O ~/.local/bin/yq
+          chmod +x ~/.local/bin/yq
+          # Replace value for filesystem permission
+          yq -i '.modules[-2].build-options.build-args[0] = "--filesystem=${{ github.workspace }}"' dev.lizardbyte.sunshine.yml
+          # Replace value for WORKSPACE variable
+          yq -i '.modules[-2].build-options.env.WORKSPACE = "${{ github.workspace }}"' dev.lizardbyte.sunshine.yml
+          sudo su $(whoami) -c '
+            flatpak run org.flatpak.Builder --arch=${{ matrix.arch }} --repo=local-repo --install-deps-from=flathub \
+            --force-clean --stop-at=cuda --default-branch=${{ github.ref_name }} --user build-sunshine dev.lizardbyte.sunshine.yml
+          '
+          # Make a copy of flatpak-builder cache before adding the CUDA module.
           cp -r .flatpak-builder copy-of-flatpak-builder
-          sudo su $(whoami) -c 'flatpak run org.flatpak.Builder --arch=${{ matrix.arch }} --repo=repo --force-clean \
-            build-sunshine dev.lizardbyte.sunshine.yml'
+          sudo su $(whoami) -c '
+            flatpak run org.flatpak.Builder --arch=${{ matrix.arch }} --repo=local-repo --install-deps-from=flathub \
+            --force-clean --default-branch=${{ github.ref_name }} --user build-sunshine dev.lizardbyte.sunshine.yml
+          '
           rm -rf .flatpak-builder
           mv copy-of-flatpak-builder .flatpak-builder
-          sudo su $(whoami) -c 'flatpak build-bundle --arch=${{ matrix.arch }} ./repo \
-            ../artifacts/sunshine_${{ matrix.arch }}.flatpak dev.lizardbyte.sunshine'
-          sudo su $(whoami) -c 'flatpak build-bundle --runtime --arch=${{ matrix.arch }} ./repo \
-            ../artifacts/sunshine_debug_${{ matrix.arch }}.flatpak dev.lizardbyte.sunshine.Debug'
+
+      - name: Bundle Flatpak
+        run: |
+          flatpak repo --branches build/local-repo
+          flatpak build-bundle build/local-repo --arch=${{ matrix.arch }} artifacts/sunshine_${{ matrix.arch }}.flatpak \
+            dev.lizardbyte.sunshine ${{ github.ref_name }}
```

Comment:
I'd prefer not to use the `${{ github.ref_name }}` as it's not consistent between PRs and push events.